spacy/tokens/span.pyx

```diff
@@ -275,12 +274,20 @@ cdef class Span:
             char_offset = self.start_char
             for key, value in self.doc.user_data.items():
                 if isinstance(key, tuple) and len(key) == 4 and key[0] == "._.":
-                    data_type, name, start, end = key
+                    data_type = key[0]
+                    name = key[1]
+                    start = key[2]
+                    end = key[3]
                     if start is not None or end is not None:
                         start -= char_offset
                         if end is not None:
                             end -= char_offset
-                        user_data[(data_type, name, start, end)] = copy.copy(value)
+                            _label = key[4]
+                            _kb_id = key[5]
+                            _span_id = key[6]
+                            user_data[(data_type, name, start, end, _label, _kb_id, _span_id)] = copy.copy(value)
+                        else:
+                            user_data[(data_type, name, start, end)] = copy.copy(value)
```

Comment:
I'm slightly confused by these nested `if`- `else` statements. Why are we reading the label/ID data conditional on the `end` not being `None`? At the very least, we should add some comments here.

Reply:
With `end not None` we ensure that it's a `Span` object because it has a `start` and an `end` position. If `end == None` then we know it's a Token. I'll add comments! 😄